hook/hooktest/service.go

```diff
@@ -0,0 +1,212 @@
+package hooktest
+
+import (
+	"sync"
+
+	"gopkg.in/errgo.v1"
+
+	"github.com/juju/gocharm/charmbits/service"
+	"github.com/juju/gocharm/hook"
+)
+
+// NewServiceFunc returns a function that can be used to create
+// new services and can be assigned to service.NewService
+// to mock the OS-level service creator.
+// When the service is started, the given Runner
+// will be used to run it (via hook.Main).
+//
+// If the notify channel is not nil, it will be used to send
+// events about services created with the function.
+// It should be buffered with a size of at least 2.
```

Comment:
If there's a requirement that the channel be buffered with a size of at least 2, would it not make sense for the channel to be created and returned by this function?